src/renderer/entities/transaction/lib/extrinsicService.ts

```diff
@@ -0,0 +1,322 @@
+import { BaseTxInfo, defineMethod, methods, OptionsWithMeta, UnsignedTransaction } from '@substrate/txwrapper-polkadot';
+import { ApiPromise } from '@polkadot/api';
+import { methods as ormlMethods } from '@substrate/txwrapper-orml';
+import { SubmittableExtrinsic } from '@polkadot/api/types';
+
+import { Transaction, TransactionType } from '@renderer/entities/transaction/model/transaction';
+import { getMaxWeight, isControllerMissing, isOldMultisigPallet } from './common/utils';
+import { IExtrinsicService } from './common/types';
+import { MultisigAccount } from '@renderer/entities/account';
+import { AccountId, Address } from '@renderer/domain/shared-kernel';
+import { toAddress } from '@renderer/shared/lib/utils';
+
+type BalancesTransferArgs = Parameters<typeof methods.balances.transfer>[0];
+type BondWithoutContollerArgs = Omit<Parameters<typeof methods.staking.bond>[0], 'controller'>;
+
+// TODO: change to substrate txwrapper method when it'll update
+const transferAllowDeath = (
+  args: BalancesTransferArgs,
+  info: BaseTxInfo,
+  options: OptionsWithMeta,
+): UnsignedTransaction =>
+  defineMethod(
+    {
+      method: {
+        args,
+        name: 'transferAllowDeath',
+        pallet: 'balances',
+      },
+      ...info,
+    },
+    options,
+  );
+
+const bondWithoutController = (
+  args: BondWithoutContollerArgs,
+  info: BaseTxInfo,
+  options: OptionsWithMeta,
+): UnsignedTransaction =>
+  defineMethod(
+    {
+      method: {
+        args,
+        name: 'bond',
+        pallet: 'staking',
+      },
+      ...info,
+    },
+    options,
+  );
+
+export const useExtrinsicService = (): IExtrinsicService => {
+  const getUnsignedTransaction: Record<
+    TransactionType,
+    (args: Transaction, info: BaseTxInfo, options: OptionsWithMeta, api: ApiPromise) => UnsignedTransaction
+  > = {
+    [TransactionType.TRANSFER]: (transaction, info, options, api) => {
+      // @ts-ignore
+      return api.tx.balances.transferAllowDeath
+        ? transferAllowDeath(
+            {
+              dest: transaction.args.dest,
+              value: transaction.args.value,
+            },
+            info,
+            options,
+          )
+        : methods.balances.transfer(
+            {
+              dest: transaction.args.dest,
+              value: transaction.args.value,
+            },
+            info,
+            options,
+          );
+    },
+    [TransactionType.ASSET_TRANSFER]: (transaction, info, options) => {
+      return methods.assets.transfer(
+        {
+          id: transaction.args.asset,
+          target: transaction.args.dest,
+          amount: transaction.args.value,
+        },
+        info,
+        options,
+      );
+    },
```

Comment:
Why do we still have all case-by-case logic here? The main goal of ExtrinsicService was to allow each flow to setup up a call or a setup of calls by its own to allow for more flexibility? I remember we decided to postpone the task for initial refactoring (which is already merged)

Reply:
Agree here. As the diagram in the pull request shows the ExtrinsicService must depend on ExtrinsicBuilder which helps to built extrinsics but still agnostic from the concrete call being used. Only modules/logic which creates an extrinsic based on user input should get the builder and set call/module name and params.